File: app/src/main/java/djordjeh/architecture/mvp/ui/BaseFragmentImpl.java
package djordjeh.architecture.mvp.ui;

import androidx.annotation.NonNull;

import javax.inject.Inject;

import dagger.android.support.DaggerFragment;

public class BaseFragmentImpl<P extends BasePresenter> extends DaggerFragment {

    @NonNull
    @Inject
    public P presenter;

    @Override
    public void onDestroyView() {
        this.presenter.stop();
        super.onDestroyView();
    }

    @Override
    public void onDestroy() {
        this.presenter.destroy();
        super.onDestroy();
    }
}

File: app/src/main/java/djordjeh/architecture/mvp/ui/BaseFragmentImpl.kt
package djordjeh.architecture.mvp.ui

import javax.inject.Inject

import dagger.android.support.DaggerFragment

open class BaseFragmentImpl<P : BasePresenter> : DaggerFragment() {

    @Inject
    lateinit var presenter: P

    override fun onDestroyView() {
        this.presenter.stop()
        super.onDestroyView()
    }

    override fun onDestroy() {
        this.presenter.destroy()
        super.onDestroy()
    }
}
